server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -329,6 +329,11 @@ protected void searchLeaf(LeafReaderContext ctx, Weight weight, Collector collec
             searchContext.setSearchTimedOut(true);
             return;
         }
+        // Resolve via star-tree
+        // if (queryStarTree(ctx, leafCollector)) {
+        // return;
+        // }
+
```

Comment:
So, in practice, you won't need any more changes in `ContextIndexSearcher` in this PR, right?